src/compiler/checker.ts

```diff
@@ -2014,6 +2017,34 @@ module ts {
             }
         }
 
+        // Push an entry on the type resolution stack. If an entry with the given target is not already on the stack,
+        // a new entry with that target and an associated result value of true is pushed on the stack, and the value
+        // true is returned. Otherwise, a circularity has occurred and the result values of the existing entry and
+        // all entries pushed after it are changed to false, and the value false is returned. The target object has
+        // no significance other than to provide a unique identity for a particular type resolution result.
+        function pushTypeResolution(target: Object): boolean {

```

Comment:
I realize the target object only matters for identity of the result. But I'd rather have one type that gets pushed onto the stack. I noticed below, the things that get pushed can be of type Symbol, SymbolLinks or Signature. I would really prefer to converge on one type if possible, just to keep things organized.